packages/@react-aria/overlays/src/calculatePosition.ts

```diff
@@ -338,9 +338,29 @@ export function calculatePositionInternal(
   delta = getDelta(crossAxis, position[crossAxis], overlaySize[crossSize], boundaryDimensions, padding);
   position[crossAxis] += delta;
 
+  // this doesn't really work great for top and bottom
+  if (crossAxis === 'left' || crossAxis === 'top') {
+    if (padding >= position[crossAxis]) {
+      position[crossAxis] = 0;
+    }
```

Comment:
Think this isn't entirely accurate because the container element could be scrollable, in which case we can get into a situation where the button is at the edge of the container but the position value is much larger than the padding and thus we don't properly readjust the popover position to be flush with the edge